### PR TITLE
WIP: new image for sccache CI

### PR DESCRIPTION
**Pull request by TriplEight**
Closes paritytech/ci_cd#73

### CODE REVIEW

**Review by drahnr — APPROVED** [review]

File: dockerfiles/sccache-ci-linux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,85 @@
+FROM ubuntu:groovy
```
Comment (drahnr) [inline dockerfiles/sccache-ci-linux/Dockerfile:1]: :+1:

File: dockerfiles/sccache-ci-linux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	CARGO_HOME=/usr/local/cargo \
+	PATH=/usr/local/cargo/bin:/usr/local/cuda-11.1/bin:$PATH \
+	CC=clang \
+	CXX=clang \
+    LD_LIBRARY_PATH=/usr/local/cuda-11.1/lib64 \
+    DEBIAN_FRONTEND=noninteractive \
+    TERM=xterm-256color
+
+# install tools and dependencies
+RUN set -eux; \
```
Comment (drahnr) [inline dockerfiles/sccache-ci-linux/Dockerfile:37]: With newer versions of podman / docker should be able to fuse the layers, so the spaghetti/chain wars should not be necessary anymore but changes can take advantage of cached layers :)

Comment (TriplEight) [reply dockerfiles/sccache-ci-linux/Dockerfile:37]: Yeah, the thing is we ignore this layer cache on the CI images builds.
And the reason is I didn't update myself about some bugs that were in docker layer caching (like if the line in Dockerfile that `COPY`es the file from local doesn't change, but the file occasionally got changed, still the old file version can be taken from the cache).

**Comment by TriplEight** [comment]
However, it's still WIP, as I didn't yet find some time to figure out what has to be configured to satisfy the dist tests.

**Comment by TriplEight** [comment]
I think it's already a good substitution for the current CI image. Further work on it will require the image to be in the CI loop.

**Pull request merged**